packages/flutter/lib/src/material/tabs.dart

```diff
@@ -901,6 +901,22 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
     return Size.fromHeight(_kTabHeight + indicatorWeight);
   }
 
+  /// Returns whether the [TabBar] contains a [Tab] with both text and icon.
+  ///
+  /// [TabBar] uses this to give uniform padding to all tabs in cases where
+  /// there are some tabs with both text and icon and some which contain only
+  /// text or icon.
+  bool get tabHasTextAndIcon {
+    for (final Widget item in tabs) {
+      if (item is Tab) {
```

Comment:
This hard codes specific knowledge of another widget into TabBar, which means the TabBar will behave differently depending on its tabs.  
This is something that is discussed in the style guide here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo/b91c3c62d96814e8fbf7119be65ddeb5a9746f82#avoid-interleaving-multiple-concepts-together

Reply:
Hmm, this is true. Although, preferredSize seems to use the same kind of code here: https://github.com/flutter/flutter/blob/f789acae4191f585f799be3c4542b16d089224a8/packages/flutter/lib/src/material/tabs.dart#L893 So I figured it would be okay to treat tabs differently to handle tabs in this special case? :thinking:

Also, this irregularity arises because the `Tabs` are treated(sized) differently within a `TabBar` in the first place.
https://github.com/flutter/flutter/blob/f789acae4191f585f799be3c4542b16d089224a8/packages/flutter/lib/src/material/tabs.dart#L106